src/components/views/game/Game.js

```diff
@@ -121,11 +125,39 @@ const Game = ({ id }) => {
         }
     }
 
-    const onClickAttack = (tile) => {
-        console.log("attack action used on tile " + tile.y + " , " + tile.x);
+    const onClickAttack = async(tile) => {
+
+        // if clicking directly on the unit, move up the path
+        if(tile.unit) {
+            for (const tile of selectedUnit.path.reverse()) {
+                selectedUnit.move(tile.x, tile.y);
+                setGameData({ ...gameData });
+                await timer(250);
+            }
+
+            //attack - arrow does not remove!
+            selectedUnit.move(tile.x, tile.y);
+            tile.unit.remove();
+            selectedUnit.showPathIndicator(false);
+            selectedUnit.showRangeIndicator(false);
```

Comment:
We need to clear everything, as some calculations are now invalid (TODO: need to do this for move too!!!)
```suggestion
            selectedUnit.showRangeIndicator(false);
            gameData.units.forEach((unit) => {
                unit.selected = false;
                unit.tilesInAttackRange = null;
                unit.tilesInAttackRangeSpecificTile = null;
                unit.traversableTiles = null;
                unit.path = null;
                unit.pathGoal = null;
            })

            setSelectedUnit(null);
```

Reply:
updated, thanks!